### PR TITLE
HPXML File Path as relative to run directory (for PAT), try 2

### PR DESCRIPTION
**Pull request by joseph-robertson**
## Pull Request Description

Currently, for the `BuildResidentialHPXML`, `BuildResidentialScheduleFile`, and `HPXMLtoOpenStudio` measures, if you specify `hpxml_path` as relative then this code gets executed:

`hpxml_path = File.expand_path(File.join(File.dirname(__FILE__), hpxml_path))`

For the purposes of collecting exported `in.xml` files, the previous line has different implications when using `OSApp` vs `PAT`:
* `OSApp`: `in.xml` is found relative to measures in `MyMeasures Directory`. Probably where you'd expect it, and where you'd want it to be.
* `PAT`: `in.xml` is found relative to measures found in the project's `analysis_xxx` measure folder. And therefore not contained in the `data_point.zip`. This is a problem because then you cannot easily download the `in.xml` files from OS-Server.

**This PR changes the above line to:**

`hpxml_path = File.expand_path(hpxml_path)`

Again, different implications for `OSApp` vs `PAT`:
* `OSApp`: Now found relative to measures in `~/AppData/.../ApplyMeasureNow/run`. You cannot use ApplyMeasureNow with BuildResidentialHPXML and then subsequently another ApplyMeasureNow with HPXMLtoOpenStudio - the second application wipes out the contents of the first. Instead, you'd use the Measures tab where you can apply both at the same time.
* `PAT`: Now found relative to the measures folder found in the data point's run folder. Therefore contained in the `data_point.zip`. Can easily download the `in.xml` files from OS-Server.

I suppose this most closely addresses item 3 from #958.

## Checklist

PR Author: Check these when they're done. Not all may apply. ~~strikethrough~~ and check any that do not apply. 

PR Reviewer: Verify each has been completed.

- [ ] Schematron validator (`EPvalidator.xml`) has been updated
- [ ] Sample files have been added/updated (via `tasks.rb`)
- [x] Unit tests have been added/updated (e.g., `HPXMLtoOpenStudio/tests`)
- [ ] Documentation has been updated
- [x] Changelog has been updated
- [x] `openstudio tasks.rb update_measures` has been run
- [ ] Checked the code coverage report on CI
- [x] No unexpected changes to simulation results of sample files

### CODE REVIEW

File: workflow/template-build-and-run-hpxml-with-stochastic-occupancy.osw [hunk 1]
```diff
@@ -83,17 +83,17 @@
     },
     {
       "arguments": {
-        "hpxml_path": "../workflow/built.xml",
+        "hpxml_path": "../../../workflow/built.xml",
         "schedules_type": "stochastic",
-        "output_csv_path": "../stochastic.csv",
-        "hpxml_output_path": "../workflow/built-stochastic-schedules.xml"
+        "output_csv_path": "../workflow/run/stochastic.csv",
+        "hpxml_output_path": "../../../workflow/built-stochastic-schedules.xml"
```
Comment (shorowit) [inline workflow/template-build-and-run-hpxml-with-stochastic-occupancy.osw:89]: This seems odd. Why do they not go up the same number of directories to get to the workflow dir?

Comment (joseph-robertson) [reply workflow/template-build-and-run-hpxml-with-stochastic-occupancy.osw:89]: `output_csv_path` is relative to the HPXML file? https://github.com/NREL/OpenStudio-HPXML/blob/relative-path-options-2/BuildResidentialScheduleFile/measure.rb#L127

Comment (shorowit) [reply workflow/template-build-and-run-hpxml-with-stochastic-occupancy.osw:89]: Shouldn't it be changed to be consistent with the other arguments?

Comment (joseph-robertson) [reply workflow/template-build-and-run-hpxml-with-stochastic-occupancy.osw:89]: Yes, but requires minor change to `BuildResidentialScheduleFile` measure.

File: BuildResidentialScheduleFile/measure.rb [hunk 2]
```diff
@@ -150,7 +150,7 @@ def create_schedules(runner, hpxml, epw_file, args)
 
     output_csv_path = args[:output_csv_path]
     unless (Pathname.new output_csv_path).absolute?
-      output_csv_path = File.expand_path(File.join(File.dirname(args[:hpxml_output_path]), output_csv_path))
+      output_csv_path = File.expand_path(output_csv_path)
```
Comment (shorowit) [inline BuildResidentialScheduleFile/measure.rb:153]: I noticed that this measure argument has some language ("Relative paths are relative to the HPXML output path") that now should be removed. Also check if any other argument descriptions need updating.

**Review by shorowit — APPROVED** [review]
This looks good to me. Thanks for figuring out a good solution here, @joseph-robertson. I'll merge this in once the CI comes back happy.

**Pull request merged**